app/resources/admin.html.template

```diff
@@ -83,23 +94,42 @@
         return false;
       }
       used_languages.push(language_code);
-      languages_to_titles[language_code] = language_title;
     }
 
     // Copy the dropdowns and the titles into the text boxes that will
     // actually be POSTed (language_menu_options and repo_titles).
     $('language_menu_options').value = JSON.stringify(used_languages);
-    $('repo_titles').value = JSON.stringify(languages_to_titles);
-
+    var languages_to_values = {};
+    for(var id_num = 0; id_num < CUSTOM_MESSAGE_TABLE_IDS.length; id_num++){
+      var table_name = CUSTOM_MESSAGE_TABLE_IDS[id_num];
+      for(var i = 0; i < used_languages.length; i++){
+        var language_value = $(table_name + '_language_value_' + (i+1)).value;
+        var used_language = used_languages[i];
+        languages_to_values[used_language] = language_value;
+      }
+      $(table_name).value = JSON.stringify(languages_to_values);
+    }
     return true;
   }
 
+  function change_text(row_number) {
```

Comment:
Can you give a better (more descriptive name) to this function? I'm trying to understand what this function does.